File: backend/src/main/java/de/ropold/backend/dto/RiskAssessmentResponse.java
package de.ropold.backend.dto;

import de.ropold.backend.model.RiskAssessmentModel;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

public record RiskAssessmentResponse(
        UUID id,
        CbcrReportBasicInfo cbcrReport,
        RiskAssessmentModel.RiskLevel overallRiskLevel,
        BigDecimal riskScore,
        Boolean lowEtrFlag,
        Boolean taxHavenFlag,
        Boolean profitShiftingIndicator,
        Boolean negativeTaxFlag,
        Boolean highRevenueLowTaxFlag,
        Boolean substanceMismatchFlag,
        Boolean blacklistJurisdictionFlag,
        BigDecimal etrVariance,
        BigDecimal revenuePerEmployeeRatio,
        BigDecimal profitPerEmployeeRatio,
        String riskExplanation,
        String recommendedAction,
        LocalDateTime createdAt,
        LocalDateTime updatedAt
) {
}

File: frontend/src/components/dto/RiskAssessmentResponse.ts
import type {CbcrReportBasicInfo} from "./CbcrReportResponse.ts";

export type RiskLevel = 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL';

export type RiskAssessmentResponse = {
    id: string;
    cbcrReport: CbcrReportBasicInfo;
    overallRiskLevel: RiskLevel;
    riskScore?: number;
    lowEtrFlag?: boolean;
    taxHavenFlag?: boolean;
    profitShiftingIndicator?: boolean;
    negativeTaxFlag?: boolean;
    highRevenueLowTaxFlag?: boolean;
    substanceMismatchFlag?: boolean;
    blacklistJurisdictionFlag?: boolean;
    etrVariance?: number;
    revenuePerEmployeeRatio?: number;
    profitPerEmployeeRatio?: number;
    riskExplanation?: string;
    recommendedAction?: string;
    createdAt: string;
    updatedAt: string;
}